File: src/main/java/com/prisonpony/assignment/adapter/persistence/jpa/LoyaltyAccount.java
package com.prisonpony.assignment.adapter.persistence.jpa;

import java.util.UUID;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;

@Entity(name = "loyalty_account")
public class LoyaltyAccount {
    @Id
    @Column(nullable = false)
    public UUID id;

    @Column(nullable = false, name = "customer_uuid")
    public UUID customerUuid;

    @Column(nullable = false)
    public int points;

    public LoyaltyAccount() {
    }

    public LoyaltyAccount(final UUID id, final UUID customerUuid, final int points) {
        this.id = id;
        this.customerUuid = customerUuid;
        this.points = points;
    }
}

File: src/main/kotlin/com/prisonpony/assignment/adapter/persistence/jpa/LoyaltyAccount.kt
package com.prisonpony.assignment.adapter.persistence.jpa

import java.util.UUID
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Id

@Entity(name = "loyalty_account")
data class LoyaltyAccount(
        @Id
        @Column(nullable = false)
        val id: UUID,

        @Column(nullable = false, name = "customer_uuid")
        val customerUuid: UUID,

        @Column(nullable = false)
        val points: Int
)
